Fixes unused MTU settings at TcpSocket dispatch

mustermeiszer opened the pull request:
This commit fixes a small bug, where the TcpSocket computed the maximum
size per payload in a tx_buffer without taking into account the MTU
defined by the underlying network device.

Commit:
Fixes unused MTU settings at TcpSocket dispatch

This commit fixes a small bug, where the TcpSocket computed the maximum
size per payload in a tx_buffer without taking into account the MTU
defined by the underlying network device.

Dirbaio commented on src/socket/tcp.rs:
```diff
@@ -1558,7 +1558,8 @@ impl<'a> TcpSocket<'a> {
                 let offset = self.remote_last_seq - self.local_seq_no;
-                let size = cmp::min(self.remote_win_len, self.remote_mss);
+                let size = cmp::min(cmp::min(self.remote_win_len, self.remote_mss),
+                     caps.max_transmission_unit);
```
I'm not convinced this is correct. MTU is IP packet size (IP header + TCP header + payload). `size` here is only payload size.

You have to subtract the IP and TCP header sizes, as in here : https://github.com/smoltcp-rs/smoltcp/blob/1268fb3d8ddeb1814a6b3a49f1647d1bbf3debd8/src/socket/tcp.rs#L1624

mustermeiszer replied:
Ah, so much for a quick fix on my part. Haha. 

Is this fine?

```rust
let size = cmp::min(cmp::min(self.remote_win_len, self.remote_mss),
                     caps.max_transmission_unit - ip_repr.buffer_len() - repr.mss_header_len());
repr.payload = self.tx_buffer.get_allocated(offset, size);
```

Dirbaio replied:
Sure!

Commit:
Taking into account TCP and IP header

The MTU consists of TCP header, IP header and the payload.
In the former fix, this was not taken into account.
